### PR TITLE
fix(unlock-app): refactored the UI when editing lock details in checkout builder

### PR DESCRIPTION
**Pull request by julien51**
# Description

A simple refactor for the checkout URL builder, in the locks section.
We now distinguish between adding a new lock and editing it more clearly.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx [hunk 1]
```diff
@@ -295,7 +283,6 @@ export const LocksForm = ({
     setLockAddress(address)
     setNetwork(config?.network)
     setDefaultValue(config ?? {})
-    setAddLock(true)
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx:298]: When editing, no need to show the form to add a new lock

File: unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx [hunk 2]
```diff
@@ -197,18 +197,6 @@ export const LocksForm = ({
             )
           }
         )}
-        <div className="flex gap-2">
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx:200]: This is moved *outside* of the list component.

File: unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx [hunk 3]
```diff
@@ -335,128 +331,142 @@ export const LocksForm = ({
...
-                  }}
-                />
+                <>
+                  <Picker
+                    userAddress={account!}
+                    onChange={(state) => {
+                      onChangeLock(state.lockAddress, state.network, state.name)
+                    }}
+                  />
+                  <Button
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx:341]: Adding a cancel button to remove the picker

**Comment by kalidiagne** [comment]
LG! 

Just on the thing, by clicking `Add a lock` when an edit is open, we don't close the lock on edit mode (settings section)

Screenshot 2023-01-19 at 13 21 52

I think that should be something like this 

Screenshot 2023-01-19 at 13 26 14

wdyt?

**Review by kalidiagne — APPROVED** [review]

**Pull request merged**